### PR TITLE
feat: Refresh rentals listings endpoint

### PR DESCRIPTION
**Pull request by LautaroPetaccio**
This PR adds the refresh rentals endpoint to refresh the rental listings when the user uses the signature to perform a rental.
Closes #11

### CODE REVIEW

**Review by fzavalia — CHANGES_REQUESTED** [review]
Good job!
Added some comments to be reviewed.

File: src/ports/rentals/component.ts [hunk 1]
```diff
@@ -109,8 +141,16 @@ export function createRentalsComponent(
     }
 
     // Verify that there's no open rental in the contract
-    const blockChainRental = await getLastBlockchainRental(rental.contractAddress, rental.tokenId)
-    if (blockChainRental && ethers.BigNumber.from(blockChainRental.endsAt).gt(fromMillisecondsToSeconds(Date.now()))) {
+    const blockChainRental = await getBlockchainRentals({
```
[the remark this answers is not in the document]
Comment (LautaroPetaccio) [reply src/ports/rentals/component.ts]: Changed it.

File: src/ports/rentals/component.ts [hunk 2]
```diff
@@ -54,20 +54,54 @@ export function createRentalsComponent(
...
+    }
+    if (options?.orderBy) {
+      querySignature += `orderBy: ${options.orderBy} `
+    }
+    if (options?.orderDirection) {
+      querySignature += `orderDirection: ${options.orderDirection} `
+    }
+    if (options?.filterBy) {
+      querySignature += `where: { ${Object.keys(options.filterBy).reduce((acc, key) => `${acc} ${key}: $${key}`, "")} }`
+      queryVariables += Object.entries(options.filterBy).reduce((acc, [key, value]) => {
```
[the remark this answers is not in the document]
Comment (LautaroPetaccio) [reply src/ports/rentals/component.ts]: The join is definitely better, I'll change it.

File: src/ports/rentals/component.ts [hunk 3]
```diff
@@ -54,20 +54,54 @@ export function createRentalsComponent(
...
+    filterBy?: Partial<BlockchainRental>
+    firsts?: number
+    orderBy?: keyof BlockchainRental
+    orderDirection?: "desc" | "asc"
+  }): Promise<BlockchainRental[]> {
+    let querySignature = ""
+    let queryVariables = ""
+
+    if (options?.firsts) {
+      querySignature += `firsts: ${options.firsts} `
```
[the remark this answers is not in the document]
Comment (LautaroPetaccio) [reply src/ports/rentals/component.ts]: It's first as you said, changing it.

File: src/controllers/handlers/rentals-handlers.ts [hunk 4]
```diff
@@ -136,3 +137,50 @@ export async function rentalsListingsCreationHandler(
...
+        status: StatusCode.NOT_FOUND,
+        body: {
+          ok: false,
+          message: error.message,
+          data: {
+            id: error.id,
+          },
+        },
+      }
+    } else if (error instanceof NFTNotFound) {
```
[the remark this answers is not in the document]
Comment (LautaroPetaccio) [reply src/controllers/handlers/rentals-handlers.ts:170]: That would work if there wasn't a throw at the end. The idea is to match any of those errors and, if not, throw the error again so it bubbles up.

File: src/components.ts [hunk 5]
```diff
@@ -16,6 +16,7 @@ export async function initComponents(): Promise<AppComponents> {
   const config = await createDotEnvConfigComponent({ path: [".env.default", ".env"] })
   const MARKETPLACE_SUBGRAPH_URL = await config.requireString("MARKETPLACE_SUBGRAPH_URL")
   const RENTALS_SUBGRAPH_URL = await config.requireString("RENTALS_SUBGRAPH_URL")
+  console.log("Marketplace subgraph url", MARKETPLACE_SUBGRAPH_URL)
```
[the remark this answers is not in the document]
Comment (LautaroPetaccio) [reply src/components.ts]: It's not! Removed it

File: src/controllers/handlers/rentals-handlers.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (fzavalia) [reply src/controllers/handlers/rentals-handlers.ts:170]:
```
if (RentalNotFound) {
  return
}

if (NFTNotFound) {
  return
}

throw
```

File: test/unit/rentals-component.spec.ts [hunk 6]
```diff
@@ -645,3 +648,250 @@ describe("when getting rental listings", () => {
...
+  describe("and there's an NFT for the given result", () => {
+    beforeEach(() => {
+      dbQueryMock.mockResolvedValueOnce({
+        rows: [rentalFromDb],
+        rowCount: 1,
+      })
+      rentalsSubgraphQueryMock.mockResolvedValueOnce({ rentals: [] })
+    })
+
+    describe("and it was updated before or equal the one in the database", () => {
```
[the remark this answers is not in the document]
Comment (LautaroPetaccio) [reply test/unit/rentals-component.spec.ts]: Splitted into two tests!

File: src/ports/rentals/component.ts [hunk 7]
```diff
@@ -109,8 +143,13 @@ export function createRentalsComponent(
     }
 
     // Verify that there's no open rental in the contract
-    const blockChainRental = await getLastBlockchainRental(rental.contractAddress, rental.tokenId)
-    if (blockChainRental && ethers.BigNumber.from(blockChainRental.endsAt).gt(fromMillisecondsToSeconds(Date.now()))) {
+    const indexerRental = await getRentalsFromIndexer({
```
Comment (fzavalia) [inline src/ports/rentals/component.ts:146]: I would name this `indexerRentals` as it is an array

**Review by fzavalia — CHANGES_REQUESTED** [review]
Just one more thing

**Review by fzavalia — APPROVED** [review]

**Pull request merged**